phoenix-core-client/src/main/java/org/apache/phoenix/schema/MetaDataClient.java

```diff
@@ -3942,11 +4013,22 @@ public MutationState dropCDC(DropCDCStatement statement) throws SQLException {
         String schemaName = statement.getTableName().getSchemaName();
         String cdcTableName = statement.getCdcObjName().getName();
         String parentTableName = statement.getTableName().getTableName();
+        String indexName = CDCUtil.getCDCIndexName(statement.getCdcObjName().getName());
+        // Mark CDC Stream as Disabled
+        long cdcIndexTimestamp = connection.getTable(indexName).getTimeStamp();
+        String streamStatusSQL = "UPSERT INTO " + SYSTEM_CDC_STREAM_STATUS_NAME + " VALUES (?, ?, ?)";
+        String streamName = String.format(CDC_STREAM_NAME_FORMAT, parentTableName, cdcTableName, cdcIndexTimestamp);
+        try (PreparedStatement ps = connection.prepareStatement(streamStatusSQL)) {
+            ps.setString(1, parentTableName);
+            ps.setString(2, streamName);
+            ps.setString(3, CDCUtil.CdcStreamStatus.DISABLED.getSerializedValue());
+            ps.executeUpdate();
+            connection.commit();
+            LOGGER.info("Marked stream {} for table {} as DISABLED", streamName, parentTableName);
+        }
```

Comment:
We should do this after the CDC drop is successful, may be even after the index drop.